client/state/followers/actions.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+const debug = require( 'debug' )( 'calypso:followers-actions' );
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	FOLLOWERS_RECEIVE,
+	FOLLOWERS_REQUEST,
+	FOLLOWERS_REQUEST_ERROR,
+	FOLLOWER_REMOVE_ERROR,
+	FOLLOWER_REMOVE_REQUEST,
+	FOLLOWER_REMOVE_SUCCESS
+} from 'state/action-types';
+
+export default {
+	fetchFollowers( query, silentUpdate = false ) {
+		return ( dispatch ) => {
+			// TODO: Componentes should not fetch if already fetching
+			debug( 'fetching followers', query );
+			if ( ! silentUpdate ) {
+				// TODO: replace FETCHING_FOLLOWERS with FOLLOWERS_REQUEST
+				dispatch( {
+					type: FOLLOWERS_REQUEST,
+					query
+				} );
+			}
+			wpcom.site( query.siteId ).statsFollowers( query, function( error, data ) {
+				// TODO: replace RECEIVE_FOLLOWERS with FOLLOWERS_RECEIVE, FOLLOWERS_REQUEST_ERROR
+				// TODO: statsFollowers should return a promise
```

Comment:
statsFollowers should already return a promise once the callback is removed, so this could be written as:
```js
wpcom.site( query.siteId ).statsFollowers( query )
	.then( data ) => dispatch( { type: FOLLOWERS_RECEIVE, query, data } )
	.catch( error ) => dispatch( { type: FOLLOWERS_REQUEST_ERROR, query, error } );
```